sklearn/utils/stats.py

```diff
@@ -40,33 +49,42 @@ def _weighted_percentile(array, sample_weight, percentile=50):
     # When sample_weight 1D, repeat for each array.shape[1]
     if array.shape != sample_weight.shape and array.shape[0] == sample_weight.shape[0]:
         sample_weight = np.tile(sample_weight, (array.shape[1], 1)).T
+
+    # Sort `array` and `sample_weight` along axis=0; set `sample_weight` for nan input
+    # to 0:
     sorted_idx = np.argsort(array, axis=0)
     sorted_weights = np.take_along_axis(sample_weight, sorted_idx, axis=0)
+    sorted_nan_mask = np.take_along_axis(np.isnan(array), sorted_idx, axis=0)
+    sorted_weights[sorted_nan_mask] = 0
 
-    # Find index of median prediction for each sample
+    # Compute the weighted cumulative distribution function (CDF) based on
+    # `sample_weight` and scale `percentile` along it:
     weight_cdf = stable_cumsum(sorted_weights, axis=0)
-    adjusted_percentile = percentile / 100 * weight_cdf[-1]
+    adjusted_percentile = percentile_rank / 100 * weight_cdf[-1]
```

Comment:
I think `adjusted_percentile` should be renamed to `adjusted_percentile_rank` for consistency.